### PR TITLE
[PM-2633] Warnings cleanup

### PR DESCRIPTION
**Pull request by withinfocus**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Cleans up some miscellaneous warnings found in the solution.

## Code changes

* **bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs:** Couple null checks to make the compiler happy, instead of the coalescing that was there before.
* **src/Api/SecretsManager/Models/Response/AccessTokenCreationResponseModel.cs:** Null markers given the use of `#nullable`; I don't really see the need to use that but it seems to be in a lot of places.
* **src/Api/SecretsManager/Models/Response/PotentialGranteeResponseModel.cs:** Opposite of the above.
* **src/Billing/Controllers/StripeController.cs:** Null checks for when a nullable is actually in use. The need for `Guid.Empty` checks is a little scary to me and potentially misguided, but they've been there a while.
* **src/Core/Tools/Models/Business/ReferenceEvent.cs:** More null markers.
* **src/Infrastructure.EntityFramework/Repositories/OrganizationDomainRepository.cs:** Use an `await` and follow the pattern used in the other methods.
* **test/Api.IntegrationTest/SecretsManager/Controllers/SecretsManagerPortingControllerTests.cs:** Unused variable.
* **test/Api.Test/Controllers/OrganizationsControllerTests.cs:** Unused repository.
* **test/Core.Test/Models/Data/SelfHostedOrganizationDetailsTests.cs:** Proper method return type since they're not asynchronous.
* **test/Infrastructure.EFIntegration.Test/Vault/Repositories/CipherRepositoryTests.cs:** Same.
* **test/IntegrationTestCommon/Factories/WebApplicationFactoryBase.cs:** Avoid clashing generic references.
* **util/EfShared/MigrationBuilderExtensions.cs:** Use an empty constant string given the use of a reference type and the subsequent operations. This is a rather fancy method to detect the up and down operations, but maintaining functionality.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs [hunk 1]
```diff
@@ -139,16 +139,22 @@ public async Task<Core.SecretsManager.Entities.Secret> UpdateAsync(Core.SecretsM
...
-                entity.Projects.Add(p);
+                if (mappedEntity.Projects != null)
+                {
+                    // Add new relationships
+                    foreach (var project in mappedEntity.Projects.Where(p => entity.Projects.All(ep => ep.Id != p.Id)))
+                    {
+                        var p = dbContext.AttachToOrGet<Project>(_ => _.Id == project.Id, () => project);
+                        entity.Projects.Add(p);
+                    }
+                }
```
Comment (Thomas-Avery) [inline bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs:157]: Should be good to just remove the null-conditional operator. 
Entity framework will return an empty collection of count 0 here, not null.

Same thing with the `mappedEntity` if the request is for removing project association.

```
  foreach (var p in entity.Projects.Where(p => mappedEntity.Projects.All(mp => mp.Id != p.Id)))
  {
      entity.Projects.Remove(p);
  }

  // Add new relationships
  foreach (var project in mappedEntity.Projects.Where(p => entity.Projects.All(ep => ep.Id != p.Id)))
  {
      var p = dbContext.AttachToOrGet<Project>(_ => _.Id == project.Id, () => project);
      entity.Projects.Add(p);
  }
```

File: bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs [hunk 2]
```diff
@@ -289,6 +295,8 @@ public async Task<(bool Read, bool Write)> AccessToSecretAsync(Guid id, Guid use
         };
 
         var policy = await query.FirstOrDefaultAsync();
+        if (policy == null)
+            return (false, false);
 
         return (policy.Read, policy.Write);
```
Comment (Thomas-Avery) [inline bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs:301]:
```suggestion

return policy == null ? (false, false) : (policy.Read, policy.Write);
```

Would you be good with this? 
It is more in line with what we have in the project repository.

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
Thanks for working on this! 

Reviewed SM related changes.

File: bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs (hunk 2 above)
Comment (withinfocus) [reply bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs:301]: Yep -- all about following norms, even this fancy one-liner 😜.

**Review by Thomas-Avery — APPROVED** [review]
SM updates LGTM!

**Review by differsthecat — APPROVED** [review]
The Vault change looks good!

**Comment by withinfocus** [comment]
Thanks everyone! We are getting close to zero warnings ...

**Pull request merged**